src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,278 @@
+        case 0: return [3]
+        case 1: return [this.absVal(size[0])]
+        default: return size
+      }
+    },
+    sign: {
+      get () {
+        return this.signSymbol
+      },
+      set (val) {
+        let i = 0
+
+        if (this.numeralPrefix) {
+          for (const char of val) {
+            if (allowedMasks['#'].test(char)) break
+            if (char === '+' || char === '-') break
+            i++
+          }
+          val = val.substr(i)
+        }
+
+        const negative = val && val.length > 0 && val[0] === '-'
+        const positiveSymbol = this.positiveSign ? '+' : ''
+
+        this.signSymbol = this.options.positiveOnly ? positiveSymbol
+          : negative ? '-' : positiveSymbol
+      }
+    },
+    numeralPrefix () {
+      return this.options.prefix ? this.options.prefix : ''
+    },
+    numeralSuffix () {
+      return this.options.suffix ? this.options.suffix : ''
+    }
+  },
+
+  methods: {
+    unmaskNumeralText (text) {
+      if (!text) return text
+
+      this.sign = text = String(text)
+      text = this.internalChange ? this.removeNonNumeral(text) : this.attemptNumeralCorrection(text)
+      if (!text) return text // attemptNumeralCorrection() may return null or falsy
+
+      let number
+      if (this.precision) {
+        number = text.split('')
+        number.splice(-this.precision, 0, '.')
+        number = this.parseFloat(number)
+      } else {
+        number = this.parseInt(text)
+      }
+
+      return this.sign + number.toString()
+    },
+    maskNumeralText (text) {
+      if (!text) return this.numeralZero()
+
+      const blockSize = Array.isArray(this.blockSize) ? this.blockSize : [this.blockSize]
+      const masked = []
+
+      let i = this.blockSeparator.length - 1
+      let j = blockSize.length - 1
+      let length = 0
+      let integer = !this.precision
+
+      text = this.sign ? String(text).substr(1) : String(text)
+
+      // Group the digits with specified separators.
+      // The first size and separator element
+      // of the arrays becomes the default value.
+      for (const digit of text.split('').reverse()) {
+        digit !== '.' && masked.push(digit)
+        if (integer && !(++length % blockSize[j])) {
+          masked.push(this.oneChar(this.blockSeparator[i], ' '))
+          i > 0 && i--
+          j > 0 && j--
+          length = 0
+        } else if (digit === '.') {
+          masked.push(this.decimal)
+          integer = true
+        }
+      }
+
+      masked.reverse()
+      // Remove leading block separator
+      this.blockSeparator.includes(masked[0]) && masked.shift()
+
+      return this.addPrefixSuffix(masked.join(''))
+    },
+    numeralZero () {
+      const zero = 0
+      this.sign = '+'
+      return this.addPrefixSuffix(zero.toFixed(this.precision).toString().replace('.', this.decimal))
+    },
+    oneChar (char, singleChar = '.') {
+      return char ? char[0] : singleChar
+    },
+    absVal (value) {
```

Comment:
`absVal` may be misleading name for function that additionaly restricts the max value

Reply:
It's enhancing `Math.abs()`. What could be a better name, anyway?

Reply:
THB I'd jsut use simply `parseInt(value, 10)` (of whatever the var name is) instead of `this.absVal(value)`. Why?

1. why anyone should provide negative numbers for block length or precision? And even if someone provide -3 for precision then probably because he want such a precision (to get only numbers like 1000, 2000, not 1234, not 1234.56) and instead he will get 1234.567

2. if someone will provide precision > 20 and we will call `toFixed` with this value - there will be clear message saying that precision > 20 is not supported, if we only limit it silently to 20, there will be no any message, and the precission will be 20 instead of intended >20. This first of all might lead to confusion and second - I bet it would be a very very rare case

Maybe I had more arguments, but was distracted and forgot. Anyway summing up - I'd just remove the method and use parseInt instead.

Reply:
1. This is a good feature. I think it could be made available next. Or should it be in this PR?
2. I think the limitation can be removed because the calculations don't use `parseInt()` or `parseFloat()` at all, what more the `toFixed()`.

Reply:
Replacing `this.absVal()` entirely with `parseInt(value, 10)` may not work because the later may return NaN. `this.absVal()` is needed to ensure that NaN is treated before proceeding.

I will proceed with item (2), but put item (1) on hold for now. `this.absVal()` may later be gone. Thank you for the suggestions.